src/typings/index.ts

```diff
@@ -1,12 +1,29 @@
-export interface IEditorElementConfig {
-  id: string;
-  type: "text";
-  visuals: Record<string, any>;
+import type { IView } from "@visactor/vgrammar";
+import type { GroupMarkVisual, RectMarkVisual, TextMarkVisual } from "./mark";
+import type { Editor } from "../editor/editor";
+
+export type IVisualConfig = IVisualTextConfig | IVisualGroupConfig | IVisualRectConfig
+
+export interface IVisualTextConfig extends IVisualBaseConfig {
+  channel: keyof GroupMarkVisual;
+}
+export interface IVisualGroupConfig extends IVisualBaseConfig {
+  channel: keyof TextMarkVisual;
+}
+export interface IVisualRectConfig extends IVisualBaseConfig {
+  channel: keyof RectMarkVisual;
 }
 
-export interface IVisualConfig {
-  channel: string;
-  type: "string" | "number" | "color" | "boolean";
-  default: any;
+export interface IVisualBaseConfig {
+  type: string | number | boolean;
+  default: string | number;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
   options?: any;
 }
+
+declare global {
```

Comment:
window.view is only used for debugging, maybe use eslint disble instead

Reply:
fixed